kedro/extras/logging/rich_logger.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import os
+import sys
+from multiprocessing.sharedctypes import Value
+from pathlib import Path
+
+import click
+import rich.logging
+import rich.pretty
+import rich.traceback
+
+logger = logging.getLogger(__file__)
+
+
+class RichHandler(rich.logging.RichHandler):
+    """Identical to rich's logging handler but with a few extra behaviours:
+    * warnings issued by the `warnings` module are redirected to logging
+    * pretty printing is enabled on the Python REPL (including IPython and Jupyter)
+    * all tracebacks are handled by rich when rich_tracebacks=True
```

Comment:
We should have anoter point here to describe the extra `rich.install` behaviour you added.